File: 240821/그 요일은/the-day-of-the-day.cpp
#include <iostream>
#include <string>
using namespace std;

string y[] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
int m[] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

int f(int a, int b){
    for(int i = 1; i < a; i++) b += m[i];
    return b;
}

int main() {
    int m1, d1, m2, d2, i = 0, c = 0;
    string a;
    cin >> m1 >> d1 >> m2 >> d2 >> a;
    d1 = f(m2, d2) - f(m1, d1);
    while(y[i] != a) i++;
    for(; i <= d1; i += 7) c++;
    cout << c;
    return 0;
}

File: 240823/그 요일은/the-day-of-the-day.py
days = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
week = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]

m1, d1, m2, d2 = map(int, input().split())
A = input()
day = (sum(days[0:m2])+d2)-(sum(days[0:m1])+d1) #1

cnt=0
cnt+=day//7 #2
if(day%7>=week.index(A)): #3
    cnt+=1

print(cnt)